src/mozilla_sec_eia/library/validation_helpers.py

```diff
@@ -65,17 +70,56 @@ def jaccard_similarity(
         value_col: Column to calculate Jaccard similarity on.
             Must be present in both dataframes.
     """
-    # fill nans to make similarity comparison more accurate
-    if (computed_df[value_col].dtype == float) and (
-        validation_df[value_col].dtype == float
-    ):
-        computed_df[value_col] = computed_df[value_col].fillna(999)
-        validation_df[value_col] = validation_df[value_col].fillna(999)
-    else:
-        computed_df[value_col] = computed_df[value_col].fillna("zzz")
-        validation_df[value_col] = validation_df[value_col].fillna("zzz")
     intersection = set(computed_df[value_col]).intersection(
         set(validation_df[value_col])
     )
     union = set(computed_df[value_col]).union(set(validation_df[value_col]))
     return float(len(intersection)) / float(len(union))
+
+
+def clean_ex21_validation_set(validation_df: pd.DataFrame):
+    """Clean Ex. 21 validation data to match extracted format."""
+    validation_df = validation_df.rename(
+        columns={
+            "Filename": "id",
+            "Subsidiary": "subsidiary",
+            "Location of Incorporation": "loc",
+            "Ownership Percentage": "own_per",
+        }
+    )
+    validation_df["own_per"] = validation_df["own_per"].astype(str)
+    validation_df["filename"] = validation_df["id"].apply(get_metadata_filename)
+    validation_df = clean_extracted_df(validation_df)
+    return validation_df
+
+
+def strip_down_company_names(ser: pd.Series) -> pd.Series:
+    """Strip LLC and other company name suffixes from a column.
+
+    Used to compare subsidiary name columns during validation.
+    """
+    # this JSON is taken from PUDL package data (used for CompanyNameCleaner)
+    json_source = (
+        resources.files("mozilla_sec_eia.package_data") / "us_legal_forms.json"
+    )
+    with json_source.open() as json_file:
+        legal_terms_dict = json.load(json_file)["legal_forms"]["en"]
+    terms_list = list(legal_terms_dict.keys())
+    for key in legal_terms_dict:
+        terms_list += legal_terms_dict[key]
+    reg = r"\b(?:" + "|".join(re.escape(word) for word in terms_list) + r")\b"
+    ser = ser.replace(reg, "", regex=True)
+    ser = ser.str.strip()
+    # strip commas or other special chars that might be at the end of the name
+    ser = ser.str.replace(r"[^[^\w&\s]+|[^\w&\s]+$]+", "", regex=True)
+    ser = ser.str.strip()
+    return ser
+
+
+def fill_nulls_for_comparison(ser: pd.Series):
```

Comment:
This is also just used for validation. The comparison metrics weren't handling nulls before and were marking them as not equal.